source/objects/responseClass.m

```diff
@@ -318,6 +318,141 @@ function plotForces(obj,bodyNum,comp)
             clear t FT FE FRD FR FV FM i
         end
         
+        function plotWaves(obj,it,tsf,g,axis_lims,body,waves,save)
+            % This method plots the wave elevation and body geometry over
+            % time to visualize the waves and response
+            %
+            % Parameters
+            % ------------
+            %     it : float
+            %         simulation time step (simu.dt)   
+            %
+            %     tsf : float
+            %         (time step factor) number of simulation time steps
+            %         per video frame (higher number decreases computation
+            %         time)
+            %
+            %     g : float
+            %         acceleration due to gravity (simu.g)
+            %
+            %     axis_lims : 1x6 float matrix
+            %         upper and lower limits to the figure
+            %         [xmin xmax ymin ymax zmin zmax]
```

Comment:
Some of these parameters are already defined in the simulation class for visualization, so you might consider requiring ``simu`` as an input instead of ``it``, `g`, `axis_lims`. Also I recommended changing your variable naming to be more informative and in camel case. I.e. ``tsf`` --> ``timesPerFrame``, ``axis_lims`` --> ``axisLimits`` (or use ``simu.domainSize`` for this)

Reply:
I'll make a PR with some of these changes directly into this file so that if you agree they are easy to pull in